dds/DCPS/XTypes/TypeObject.h

```diff
@@ -11,22 +11,41 @@
 #include "dds/DCPS/PoolAllocator.h"
 #include "dds/DCPS/Serializer.h"
 
-#include <tao/Array_VarOut_T.h>
-
 #include <ace/CDR_Base.h>
 
 #include <algorithm>
 #include <cstring>
 
 OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
-
-
-namespace DDS {
-  class OctetSeq;
-}
 namespace OpenDDS {
 namespace XTypes {
 
+  template<typename T, typename T_slice, typename TAG>
+  class Fake_TAO_Array_Forany_T {
```

Comment:
Rename?  Reformat?

Reply:
Marking this as "resolved" since we will handle it in a separate PR.

The approach I'm suggesting is to use `IDL::DistinctType<T>` from `Serializer.h`.  I think that change could be made here without impacting other uses.  We could probably do it when true mutable/appendable support is added for these types (Real Soon Now).